evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py

```diff
@@ -385,50 +480,114 @@ def _decompose_target(X, y, fh):
             )
 
         if isinstance(y, pd.Series):
-            result_dfs.append(_decompose_target(X, y, None))
-        elif isinstance(y, pd.DataFrame):
-            for colname in y.columns:
-                result_dfs.append(_decompose_target(X, y[colname], None))
-        return result_dfs
+            y = y.to_frame()
+        series_results = {}
+        # Iterate through each series id
+        for id in y.columns:
+            result_dfs = []
+            if not isinstance(X.index, pd.DatetimeIndex):
+                raise TypeError("Provided X should have datetimes in the index.")
+            if X.index.freq is None:
+                raise ValueError(
+                    "Provided DatetimeIndex of X should have an inferred frequency.",
+                )
```

Comment:
If I reset the indexes then it causes problems so I think this check `if not isinstance(X.index, pd.DatetimeIndex)` is necessary but removing the frequency didn't do anything so I'm not sure if checking the index frequency is necessary

Reply:
Cool, let's remove the frequency check while we're here.

The index check is a pet peeve of mine, I run into it so often - can we remove it by manually setting X's index to y's index, since we already manually set y's index to be a DatetimeIndex? This isn't a requirement, just trying to save myself some frustration 😅

Reply:
Done! Should this be changed in the `PolynomialDecomposer` as well?